### PR TITLE
fix(reify): cleanup of Symbols

### PR DESCRIPTION
**Pull request by wraithgar**
This starts the long process of cleaning reify.js up.  All of the
remaining symbols are used either elsewhere in arborist, or in tests.

A single use function was inlined, and `dedupe` was moved to the main
arborist class.

### CODE REVIEW

File: workspaces/arborist/lib/arborist/reify.js [hunk 1]
```diff
@@ -169,7 +146,124 @@ module.exports = cls => class Reifier extends cls {
...
+      await this.actualTree.meta.save()
+      const ignoreScripts = !!this.options.ignoreScripts
+      // if we aren't doing a dry run or ignoring scripts and we actually made changes to the dep
+      // tree, then run the dependencies scripts
+      if (!this.options.dryRun && !ignoreScripts && this.diff && this.diff.children.length) {
+        const { path, package: pkg } = this.actualTree.target
+        const stdio = this.options.foregroundScripts ? 'inherit' : 'pipe'
+        const { scripts = {} } = pkg
+        for (const event of ['predependencies', 'dependencies', 'postdependencies']) {
+          if (Object.prototype.hasOwnProperty.call(scripts, event)) {
```
Comment (ljharb) [inline workspaces/arborist/lib/arborist/reify.js:254]:
```suggestion
          if (Object.hasOwn(scripts, event)) {
```
supported in node 16+

Comment (wraithgar) [reply workspaces/arborist/lib/arborist/reify.js:254]: We're not shaving the whole yak here.

Comment (ljharb) [reply workspaces/arborist/lib/arborist/reify.js:254]: sure, just a suggestion

**Review by lukekarrys — APPROVED** [review]
This represents a ton of work that will be very beneficial going forward! 👏🏼

**Pull request merged**